### PR TITLE
Erce/minimum basefee

### LINKED ISSUE #998: Set a minimum value for base fee
Currently we are using eip-1559 for fee calculations, however when there are a lot of empty blocks in a row the base fee drops down rapidly and it is hard to get it back up. That's why we can set a minimum value to our base fee like [arbitrum](https://docs.arbitrum.io/how-arbitrum-works/gas-fees#gas-price-floor) did.

### PR DESCRIPTION
**Pull request by ercecan**
# Description
Sets a minimum value to basefee

## Linked Issues
- Fixes #998

### CODE REVIEW

File: crates/sovereign-sdk/fuzz/Cargo.lock
Comment (kpp) [inline crates/sovereign-sdk/fuzz/Cargo.lock:1]: Why is it so big?

File: crates/primitives/src/basefee.rs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+use crate::MIN_BASE_FEE_PER_GAS;
+
+pub fn calculate_next_block_base_fee(
+    gas_used: u128,
+    gas_limit: u128,
+    base_fee: Option<u64>,
+    base_fee_params: BaseFeeParams,
+) -> Option<u64> {
+    Some(std::cmp::max(
```
Comment (eyusufatik) [inline crates/primitives/src/basefee.rs:12]: why option?

+ shouldn't this return u128?

File: bin/citrea/tests/evm/fee.rs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    assert!(block.header.base_fee_per_gas.unwrap() >= 10000000);
+    for _ in 0..100 {
+        test_client.spam_publish_batch_request().await.unwrap();
+    }
+
+    let block = test_client
+        .eth_get_block_by_number(Some(BlockNumberOrTag::Latest))
+        .await;
+    // Base fee should at most be 0.01 gwei
+    assert!(block.header.base_fee_per_gas.unwrap() >= 10000000);
```
Comment (eyusufatik) [inline bin/citrea/tests/evm/fee.rs:57]: make this assert base_fee == 10000000

**Review by eyusufatik — CHANGES_REQUESTED** [review]

File: crates/sovereign-sdk/fuzz/Cargo.lock
Comment (ercecan) [reply crates/sovereign-sdk/fuzz/Cargo.lock:1]: added reth primitives to primitives, and it is big

File: crates/primitives/src/basefee.rs (hunk 1 above)
Comment (ercecan) [reply crates/primitives/src/basefee.rs:12]: The old way we calc basefee is usually something like this:
```rust
parent_block
.header
.next_block_base_fee(cfg.base_fee_params)
```
the function  next_block_base_fee returns `Option<u64>` that's why I tried to make it sth like that

also  pretty much everywhere basefee is u64

Comment (eyusufatik) [reply crates/primitives/src/basefee.rs:12]: ok, I'll respect that

**Review by eyusufatik — APPROVED** [review]

**Pull request merged**